src/Config.php

```diff
@@ -70,10 +69,92 @@ public function isAvailable()
      * @param string|null $envPrefix
      *   The prefix for environment variables. Defaults to 'PLATFORM_'.
      */
-    public function __construct(array $environmentVariables = null, $envPrefix = null)
+    public function __construct(array $environmentVariables = null, string $envPrefix = null)
+    {
+        $this->environmentVariables = $environmentVariables ?? $this->getEnv();
+        $this->envPrefix = $envPrefix ?? 'PLATFORM_';
+
+        if ($this->isAvailable() && !$this->inBuild() && isset($this->environmentVariables['PLATFORM_ROUTES'])) {
+            $this->routes = $this->decode($this->getValue('ROUTES'));
+        }
+    }
+
+    /**
+     * Checks whether any configuration is available.
+     *
+     * @return bool
+     *   True if configuration can be used, false otherwise.
+     */
+    public function isAvailable() : bool
+    {
+        return isset($this->environmentVariables[$this->envPrefix . 'APPLICATION']);
+    }
+
+    /**
+     * Checks whether the code is running in a build environment.
+     *
+     * If false, it's running at deploy time.
+     *
+     * @return bool
+     */
+    public function inBuild() : bool
+    {
+        return $this->isAvailable() && !$this->getValue('ENVIRONMENT');
+    }
+
+    /**
+     * Returns the routes definition.
+     *
+     * @return array
+     *   The routes array, in PHP nested array form.
+     * @throws \RuntimeException
+     *   If the routes are not accessible due to being in the wrong environment.
+     */
+    public function routes() : array
+    {
+        if (!$this->isAvailable()) {
+            throw new \RuntimeException('You are not running on Platform.sh, so routes are not available.');
+        }
+
+        if ($this->inBuild()) {
+            throw new \RuntimeException('Routes are not available during the build phase.');
+        }
+
+        return $this->routes;
+    }
+
+    /**
+     * @param string $id
+     *   The ID of the route to load.
+     * @return array
+     *   The route definition.  The generated URL of the route is added as a "url" key.
+     * @throws \InvalidArgumentException
+     *   If there is no route by that ID, an exception is thrown.
+     */
+    public function getRoute(string $id) : array
+    {
+        foreach ($this->routes() as $url => $route) {
+            if ($route['id'] == $id) {
+                // This is non-standard, but means we don't need to return the array key in parallel.
+                $route['url'] = $url;
+                return $route;
+            }
```

Comment:
This logic implies that a route returned by ID has a different structure than a route found by the routes() method.  I am not a fan of this, but I am not sure if there's a good way to address it other than modifying the API to include the URL to begin with.  (Which I'd support, frankly.)

Also, it really pains me to use an array here rather than a classed object, but I don't know if that's worth the effort.

Feedback welcome here.

Reply:
I think this might be faster:
`$routes = $this->routes();
$key = array_search($id, array_column($routes, 'id'));
return $routes[$key]["url"];`

Reply:
Almost.  The problem is array_column() returns a list of ids in an ordered array, not associative, and then array_search() returns the numeric key of that array, which is not how $routes is keyed.  (It'e keyed by url.)  So the search always fails.